### PR TITLE
Feature/dynamic fields

### PR DESCRIPTION
**Pull request by LMDoc**
Now supports any collection regardless of required fields. Fixed security issue in dependency. Added to readme for more clarity regarding supported types.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -217,12 +243,12 @@ Resizer.prototype.changeSize = function (itemDetails) {
...
     let myBucket = this.config.aws.bucket;
-    let myKey = `${itemDetails.itemId}-${itemDetails.fieldId}.jpg`;
-    let params = { Bucket: myBucket, Key: myKey, Body: img };
+    let myKey = `${itemDetails.itemId}-${itemDetails.fieldId}.png`;
+    let params = {Bucket: myBucket, Key: myKey, Body: img};
 
     return this.s3.putObject(params, function (err, data) {
         if (err) {
-            console.error(
+            console.log(
```
Comment (wivern-co-uk) [inline index.js]: Why it's changed to `log`?

File: index.js [hunk 2]
```diff
@@ -200,14 +226,14 @@ Resizer.prototype.changeSize = function (itemDetails) {
         .then(image => {
             return image
                 .scaleToFit(itemDetails.width, itemDetails.height) // resize
-                .quality(100) // set JPEG quality     
-                .getBufferAsync(Jimp.MIME_JPEG)
+                .quality(this.config.quality) // set PNG quality
```
Comment (wivern-co-uk) [inline index.js]: PNG don't have a quality like JPEG, something is wrong here

File: index.js [hunk 3]
```diff
@@ -200,14 +226,14 @@ Resizer.prototype.changeSize = function (itemDetails) {
...
+                .getBufferAsync(Jimp.MIME_PNG)
```
Comment (wivern-co-uk) [inline index.js]: Does it mean all images will be converted to PNG? Shouldn't it preserve the original format?

**Review by wivern-co-uk — CHANGES_REQUESTED** [review]

File: index.js (hunk 3 above)
Comment (LMDoc) [reply index.js]: I believe JIMP can get set to use the original format, I'll look into this

File: index.js [hunk 4]
```diff
@@ -198,10 +224,17 @@ Resizer.prototype.validateImageSize = function (itemDetails) {
 Resizer.prototype.changeSize = function (itemDetails) {
     return Jimp.read(itemDetails.imageUrl)
         .then(image => {
+            const mime =
+                image.getMIME().split("/")[1] === "jpg" || "png" || "bmp"
```
Comment (wivern-co-uk) [inline index.js]: It works not as expected:
```
"jpg" === "jpg" || "png" || "bmp"
> true
"png" === "jpg" || "png" || "bmp"
> "png"
"bmp" === "jpg" || "png" || "bmp"
> "png"
"nonsense" === "jpg" || "png" || "bmp"
> "png"
```

**Review by wivern-co-uk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**